tests/test_etcd3.py

```diff
@@ -88,6 +89,90 @@ def test_delete_key(self, etcd):
         with pytest.raises(etcd3.exceptions.KeyNotFoundError):
             etcd.get('/doot/delete_this')
 
+    def test_watch_key(slef, etcd):
+        def update_etcd(v):
+            etcdctl('put', '/doot/watch', v)
+            out = etcdctl('get', '/doot/watch')
+            assert base64.b64decode(out['kvs'][0]['value']).decode('utf-8') == v
+
+        def update_key():
+            # sleep to make watch can get the event
+            time.sleep(3)
+            update_etcd('1')
+            time.sleep(1)
+            update_etcd('2')
+            time.sleep(1)
+            update_etcd('3')
+            time.sleep(1)
+            update_etcd('4')
+            time.sleep(1)
+
+        t = threading.Thread(name="update_key", target=update_key)
+        t.start()
+
+        key_change_count = 0
+        for (event, cancel) in etcd.watch('/doot/watch'):
+            if key_change_count == 0: # first time will not have any events
+                assert event.created is True
+                assert len(event.events) == 0
+            else:
+                assert event.created is False
+                assert len(event.events) == 1
+                assert event.events[0].kv.value.decode('utf-8') == str(key_change_count)
```

Comment:
Same here, I think this should be something more similar to

    assert event.events[0].kv.value == utils.to_bytes(key_change_count)